themes/_core/js/product.js

```diff
@@ -27,125 +27,156 @@ import prestashop from 'prestashop';
+                    '&preview=' + preview,
+                method: 'POST',
+                data: {
+                    ajax: 1,
+                    action: 'refresh',
+                    quantity_wanted: $quantityWantedInput.val()
+                },
+                dataType: 'json',
+                beforeSend: function() {
+                    if (currentRequest != null) {
+                        currentRequest.abort();
+                    }
+                },
+                error: function(jqXHR, textStatus, errorThrown) {
+                    if (textStatus !== 'abort'
+                        && $('section#main > .ajax-error').length === 0
+                    ) {
+                        showError($('#product-availability'), 'An error occurred while processing your request');
+                    }
+                },
+                success: function(data, textStatus, errorThrown) {
+                    // Avoid image to blink each time we modify the product quantity
+                    // Can not compare directly cause of HTML comments in data.
+                    const $newImagesContainer = $('<div>').append(data.product_cover_thumbnails);
+
+                    // Used to avoid image blinking if same image = epileptic friendly
+                    if ($('.images-container').html() !== $newImagesContainer.find('.images-container').html()) {
+                        $('.images-container').replaceWith(data.product_cover_thumbnails);
+                    }
+                    $('.product-prices').replaceWith(data.product_prices);
+                    $('.product-customization').replaceWith(data.product_customization);
+                    $('.product-variants').replaceWith(data.product_variants);
+                    $('.product-discounts').replaceWith(data.product_discounts);
+                    $('.product-additional-info').replaceWith(data.product_additional_info);
+                    $('#product-details').replaceWith(data.product_details);
+                    replaceAddToCartSections(data);
+                    const minimalProductQuantity = parseInt(data.product_minimal_quantity, 10);
+
+                    // Prevent quantity input from blinking with classic theme.
+                    if (!isNaN(minimalProductQuantity)
+                        && $quantityWantedInput.val() < minimalProductQuantity
+                        && eventType !== 'updatedProductQuantity'
+                    ) {
+                        $quantityWantedInput.attr('min', minimalProductQuantity);
+                        $quantityWantedInput.val(minimalProductQuantity);
+                    }
+                    prestashop.emit('updatedProduct', data);
+                },
+                complete: function(jqXHR, textStatus) {
+                    currentRequest = null;
+                    currentRequestDelayedId = null;
+                }
+            });
+        }.bind(currentRequest, currentRequestDelayedId), 250);
```

Comment:
ok for the delay, but do we need to use setTimeout for it? can't we setup a delay directly in ajax call ?